### PR TITLE
feat: Fir 24158 new identity resource manager

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Updated ResourceManager to use a new SQL api to handle resources
- deprecated Settings usage in favor of passing parameter seprately
- removed binding, region, provider and engine_revision models and services
- fixed unit and integration tests

### CODE REVIEW

File: tests/integration/resource_manager/test_engine.py [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
 
 
 def make_engine_name(database_name: str, suffix: str) -> str:
     return f"{database_name}_{suffix}"
 
 
-@pytest.mark.skip(reason="manual test")
-def test_create_start_stop_engine(database_name: str):
-    rm = ResourceManager()
+# @pytest.mark.skip(reason="manual test")
```
Comment (ptiurin) [inline tests/integration/resource_manager/test_engine.py:12]: leftover comment

**Review by ptiurin — COMMENTED** [review]
Also, unit tests are kaput :'(

File: src/firebolt/model/engine.py [hunk 2]
```diff
@@ -193,300 +124,124 @@ def get_connection(self) -> Connection:
...
-                        f"did not stop within {wait_timeout_seconds} seconds."
-                    ),
-                    verbose=True,
-                )
-                engine = engine.get_latest()
 
-            logger.info(
-                f"Engine (engine_id={engine.engine_id}, name={engine.name}) stopped."
+        self.refresh()
+        self._wait_for_start_stop()
```
[the remark this answers is not in the document]
Comment (stepansergeevitch) [reply src/firebolt/model/engine.py:163]: This is just preserving a previous behaviour. I think we had cases where this behaviour was needed and we added this.

File: src/firebolt/model/engine.py [hunk 3]
```diff
@@ -99,90 +42,78 @@ def inner(self: Engine, *args: Any, **kwargs: Any) -> Any:
...
+                pass
+        return None
+
+    def refresh(self) -> None:
+        """Update attributes of the instance from Firebolt."""
+        field_name_overrides = self._get_field_overrides()
+        for name, value in self._service._get_dict(self.name).items():
+            setattr(self, field_name_overrides.get(name, name), value)
+
+        self.__post_init__()
```
[the remark this answers is not in the document]
Comment (stepansergeevitch) [reply src/firebolt/model/engine.py:107]: We just fetch the engine state from db and update each class field with a new value.
Then we do post init for value post processing

File: src/firebolt/service/database.py [hunk 4]
```diff
@@ -1,53 +1,59 @@
...
+    )
+    GET_SQL = f"SELECT {', '.join(DB_FIELDS)} FROM information_schema.databases"
+    GET_BY_NAME_SQL = GET_SQL + " WHERE database_name=?"
+    GET_WHERE_SQL = " WHERE "
+
+    CREATE_PREFIX_SQL = "CREATE DATABASE {}{}"
+    CREATE_WITH_SQL = " WITH "
+    IF_NOT_EXISTS_SQL = "IF NOT EXISTS "
+    CREATE_PARAMETER_NAMES = (
+        "REGION",
```
[the remark this answers is not in the document]
Comment (stepansergeevitch) [reply src/firebolt/service/database.py:33]: SQL allows you to provide this, that's why it's here. I'm not sure, maybe there are cases where you need to create a db in a different region

**Review by ptiurin — APPROVED** [review]
Apart from that LGTM

**Pull request merged**